nbcparser/main.go

```diff
@@ -1,40 +1,55 @@
 package main
 
 import (
+	"flag"
 	"log"
 	"os"
+	"os/exec"
 
 	"github.com/Azure/agentbaker/nbcparser/pkg/parser"
 )
 
-// input to this function will be the serialized JSON from userdata + custom data.
-// it will be deserialized to the contract that the VHD this binary will be on supports.
-// Parse will be called using that deserialized struct and output the generated cse_cmd to trigger the bootstrap process.
-// example usage:
-// to build: go build main.go.
-// to run: ./main testdata/test_nbc.json.
-//
-//nolint:gosec // generated cse_cmd.sh file needs execute permissions for bootstrapping
+// This script will take in the bootstrap contract in JSON format,
+// it will be deserialized to the contract that the VHD this binary supports.
+// The contract will be used to generate cse_cmd and trigger the bootstrap process.
 func main() {
-	if len(os.Args) < parser.MinArgs {
-		log.Default().Printf("Usage: %s <input.json>", os.Args[0])
-		return
+	configFile := flag.String("bootstrap-config", "", "bootstrap configuration json filepath")
+	test := flag.Bool("test", false, "test mode")
+	flag.Parse()
+	if *configFile == "" {
+		log.Fatal("bootstrap-config is a required argument")
 	}
 
-	// Read in the JSON file
-	inputJSON, err := os.ReadFile(os.Args[1])
+	cseCmd := parseConfig(*configFile)
+	err := bootstrap(cseCmd, *test)
 	if err != nil {
 		log.Fatal(err)
-		return
 	}
+	log.Println("Provisioning completed successfully")
```

Comment:
I could be wrong. Is it more accurate to say Provisioning request initiated(received)? We don't know if it's successful or failed until the scripts really complete, right?

Reply:
Right, im just going to remove the print statement altogether since it doesn't tell us much. The cse script to poll the status of the systemd unit will provide information on whether bootstrapping completed successfully.